math/src/elliptic_curve/short_weierstrass/curves/bls12_377/curve.rs

```diff
@@ -12,8 +14,28 @@ const BLS12377_MAIN_SUBGROUP_ORDER: U384 =
 /// The description of the curve.
 #[derive(Clone, Debug)]
 pub struct BLS12377Curve;
-impl IsShortWeierstrass for BLS12377Curve {
+
+impl IsEllipticCurve for BLS12377Curve {
     type BaseField = BLS12377PrimeField;
+    type PointRepresentation = ProjectivePoint<Self>;
+
+    fn generator() -> Self::PointRepresentation {
```

Comment:
I think that it'd be a good idea for the `generator` function should be `const` in the future. We can file an issue for this.

Reply:
Or at least to have a `const` version.